samtranslator/model/__init__.py

```diff
@@ -434,7 +434,8 @@ def _resolve_string_parameter(self, intrinsics_resolver, parameter_value, parame
         if not parameter_value:
             return parameter_value
         value = intrinsics_resolver.resolve_parameter_refs(parameter_value)
-        if not isinstance(value, string_types):
+
+        if not isinstance(value, string_types) and not isinstance(value, dict):
```

Comment:
When resolving intrinsics, if SAM can't or won't resolve all of them, it still returns an intrinsic dictionary. The previous version of this function did not allow for a dictionary to be returned from the intrinsic resolver.

Reply:
So are string and dictionary the only 2 cases that will be allowed?

Reply:
Yes. Correct values here are either a string or an intrinsic function, which is interpreted as a dictionary.